content/shmem_and.tex

```diff
@@ -0,0 +1,44 @@
+\apisummary{
+  Atomically perform a non-fetching bitwise and operation on a
+  remote data object.
+}
+
+\begin{apidefinition}
+
+\begin{C11synopsis}
+void shmem_and(TYPE *dest, TYPE value, int pe);
+\end{C11synopsis}
+where \TYPE{} is one of the standard \ac{AMO} types specified by
+Table~\ref{stdamotypes}.
```

Comment:
The C standard says that signed integer types (&sect;6.2.5-4) have an _implementation-defined_ representation (&sect;6.2.6.2-2), so I don't think OpenSHMEM is overstepping and defining something that is undefined by the language. This would only seem to be problematic if the signed integer representations of the CPU and NIC did not match, which seems unlikely.

In general, I doubt we'll run into much trouble with regards to signed integer representation. Quoting [Wikipedia](https://en.wikipedia.org/wiki/Signed_number_representations#History):
> [...] As IC technology advanced, virtually all [processors] adopted two's complement technology. x86, m68k, Power Architecture, MIPS, SPARC, ARM, Itanium, PA-RISC, and DEC Alpha processors are all two's complement.